packages/styled-components/src/types.ts

```diff
@@ -230,15 +237,25 @@ export interface IInlineStyle<Props extends object> {
   generateStyleObject(executionContext: ExecutionContext & Props): object;
 }
 
-export type StyledObject<Props extends object> = CSS.Properties<number | (string & {})> & {
-  [key: string]:
-    | string
-    | number
-    | StyleFunction<Props>
-    | StyledObject<Props>
-    | RuleSet<any>
-    | undefined;
-};
+export type CSSProperties = CSS.Properties<number | (string & {})>;
```

Comment:
Last thing is these types need to be exposed in the barrel files `src/index.ts` and `src/native/index.ts`

Reply:
Thanks, I completely missed that.

And I added it.
https://github.com/styled-components/styled-components/pull/4117/commits/765498df6975b20ae52b04ef6b3b77ac78098f91